packages/venia-ui/lib/components/App/localeProvider.js

```diff
@@ -39,6 +53,8 @@ const LocaleProvider = props => {
         }
     };
 
+    if (loading) return fullPageLoadingIndicator;
```

Comment:
I know it was causing a warning but I'd like to try and figure out why before we just merge this line. If you change it to 

```js
if (loading && !data) {
   debugger;
  return fullPageLoadingIndicator
}
```

does it pause after you change the locales?

If so, it means that for some reason data is getting deleted even after you refresh the page,

You might also try changing this to `if (!messages)` which should be the same thing.

Reply:
It doesn't step into condition `if (loading && !data)` and stop on debugger when I switch locales. Since we already have `data` cached `storeConfig` for all stores in `storeSwitcher`

It does stop on `if (loading && !messages)` and just `if (!messages)`, but still got warning.

Leaving just `if (loading)` works without warning

Reply:
OK. That's fine. No need to debug the deeper issue with the warning right now.